docs/build_pyodide_packages_list.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+import sys
+import pathlib
+
+base_dir = pathlib.Path(__file__).resolve().parent.parent
+sys.path.append(str(base_dir / "pyodide-build"))
+
+from pyodide_build.io import parse_package_config
+
+PACKAGES_DIR = base_dir / "packages"
+OUTPUT_FILE = "packages-in-pyodide.md"
+OUTPUT_PATH = base_dir / "docs" / "usage" / OUTPUT_FILE
+
+TEMPLATE = """
+(packages-in-pyodide)=
+# Packages built in Pyodide
+
+The list of prebuilt python packages in pyodide.
+These packages can be loaded through {{any}}`pyodide.loadPackage` or {{any}}`micropip.install`.
+See {{ref}}`loading_packages` for information about loading packages.
+
+
+| Name  | Version |
+| :---: | :-----: |
+{packages}
+"""
+
+
+def parse_package_info(config):
+    yaml_data = parse_package_config(config)
+
+    name = yaml_data["package"]["name"]
+    version = yaml_data["package"]["version"]
+    is_library = yaml_data.get("build", {}).get("library", False)
+
+    return name, version, is_library
+
+
+def get_package_metadata_list(directory: pathlib.Path):
+    return directory.glob("**/meta.yaml")
+
+
+def to_markdown(template, packages):
```

Comment:
Could you please add type annotations everywhere, for function input arguments and output? That would help readability particularly in the case where there are no docstrings (which is fine here).